### ISSUE TITLE
Couldn't read row ... Make sure the Cursor is initialized at `createFileInstance`

### ISSUE THREAD

**Issue by JanisPlayer** [nextcloud/android#9948]
************ CAUSE OF ERROR ************

java.lang.IllegalStateException: Couldn't read row 971, col 3 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
	at android.database.CursorWindow.nativeGetString(Native Method)
	at android.database.CursorWindow.getString(CursorWindow.java:469)
	at android.database.AbstractWindowedCursor.getString(AbstractWindowedCursor.java:53)
	at android.database.CursorWrapper.getString(CursorWrapper.java:141)
	at com.owncloud.android.datamodel.FileDataStorageManager.createFileInstance(FileDataStorageManager.java:988)
	at com.owncloud.android.datamodel.FileDataStorageManager.getFolderContent(FileDataStorageManager.java:915)
	at com.owncloud.android.datamodel.FileDataStorageManager.getFolderContent(FileDataStorageManager.java:185)
	at com.owncloud.android.ui.adapter.OCFileListAdapter.swapDirectory(OCFileListAdapter.java:837)
	at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1236)
	at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1188)
	at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1177)
	at com.owncloud.android.ui.activity.FileDisplayActivity.updateListOfFilesFragment(FileDisplayActivity.java:687)
	at com.owncloud.android.ui.activity.FileDisplayActivity$UploadFinishReceiver.onReceive(FileDisplayActivity.java:1451)
	at androidx.localbroadcastmanager.content.LocalBroadcastManager.executePendingBroadcasts(LocalBroadcastManager.java:313)
	at androidx.localbroadcastmanager.content.LocalBroadcastManager$1.handleMessage(LocalBroadcastManager.java:121)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:233)
	at android.app.ActivityThread.main(ActivityThread.java:8030)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30190090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: OnePlus
Device: OnePlus8T
Model: KB2003
Id: RP1A.201005.001
Product: OnePlus8T_EEA

************ FIRMWARE ************
SDK: 30
Release: 11
Incremental: 2202182344

**Comment by AlvaroBrey** [comment]
Using this issue to centralize all other crashes with a similar stacktrace at `createFileInstance`

**Comment by 0x7c9** [comment]
when will be solution ready? I want use app :) I like it! how i can help with dev?

**Comment by fabioleite0** [comment]
Same here:

### Cause of error
```java
Exception in thread "main" java.lang.IllegalStateException: Couldn't read row 713, col 3 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
    at android.database.CursorWindow.nativeGetString(Native Method)
    at android.database.CursorWindow.getString(CursorWindow.java:462)
    at android.database.AbstractWindowedCursor.getString(AbstractWindowedCursor.java:54)
    at android.database.CursorWrapper.getString(CursorWrapper.java:141)
    at com.owncloud.android.datamodel.FileDataStorageManager.createFileInstance(FileDataStorageManager.java:967)
    at com.owncloud.android.datamodel.FileDataStorageManager.getGalleryItems(FileDataStorageManager.java:2234)
    at com.owncloud.android.datamodel.FileDataStorageManager.getAllGalleryItems(FileDataStorageManager.java:2179)
    at com.owncloud.android.ui.adapter.GalleryAdapter.showAllGalleryItems(GalleryAdapter.kt:171)
    at com.owncloud.android.ui.fragment.GalleryFragment.showAllGalleryItems(GalleryFragment.java:348)
    at com.owncloud.android.ui.fragment.GalleryFragment.handleSearchEvent(GalleryFragment.java:192)
    at com.owncloud.android.ui.fragment.GalleryFragment.onRefresh(GalleryFragment.java:162)
    at androidx.swiperefreshlayout.widget.SwipeRefreshLayout$1.onAnimationEnd(SwipeRefreshLayout.java:195)
    at androidx.swiperefreshlayout.widget.CircleImageView.onAnimationEnd(CircleImageView.java:107)
    at android.view.ViewGroup.finishAnimatingView(ViewGroup.java:7134)
    at android.view.View.draw(View.java:22613)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.draw(View.java:22707)
    at android.view.View.updateDisplayListIfDirty(View.java:21579)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.updateDisplayListIfDirty(View.java:21570)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.updateDisplayListIfDirty(View.java:21570)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.draw(View.java:22707)
    at android.view.View.updateDisplayListIfDirty(View.java:21579)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at androidx.coordinatorlayout.widget.CoordinatorLayout.drawChild(CoordinatorLayout.java:1277)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.updateDisplayListIfDirty(View.java:21570)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at androidx.drawerlayout.widget.DrawerLayout.drawChild(DrawerLayout.java:1478)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.draw(View.java:22707)
    at android.view.View.updateDisplayListIfDirty(View.java:21579)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.updateDisplayListIfDirty(View.java:21570)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.updateDisplayListIfDirty(View.java:21570)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.updateDisplayListIfDirty(View.java:21570)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.updateDisplayListIfDirty(View.java:21570)
    at android.view.View.draw(View.java:22435)
    at android.view.ViewGroup.drawChild(ViewGroup.java:4528)
    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4289)
    at android.view.View.draw(View.java:22707)
    at com.android.internal.policy.DecorView.draw(DecorView.java:819)
    at android.view.View.updateDisplayListIfDirty(View.java:21579)
    at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:534)
    at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:540)
    at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:616)
    at android.view.ViewRootImpl.draw(ViewRootImpl.java:4531)
    at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:4251)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3374)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2179)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8793)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1037)
    at android.view.Choreographer.doCallbacks(Choreographer.java:845)
    at android.view.Choreographer.doFrame(Choreographer.java:780)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:201)
    at android.os.Looper.loop(Looper.java:288)
    at android.app.ActivityThread.main(ActivityThread.java:7870)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30210090`
* Build flavor: `gplay`

### Device information
* Brand: `google`
* Device: `bramble`
* Model: `Pixel 4a (5G)`
* Id: `SQ3A.220705.003.A1`
* Product: `bramble`

### Firmware
* SDK: `32`
* Release: `12`
* Incremental: `8672226`

**Comment by Cebrain** [comment]
Any Update on this Issue?
I have 2 Smartphones, one Devices crashes when i access the media tab. On the other phone everything works fine.

One thing i can see is, that the thumbnails on the crashing phone is messed up.
I did a reset of the app already.

**Comment by AlvaroBrey** [comment]
> Any Update on this Issue?

See https://github.com/nextcloud/android/issues/10213 for the challenges in fixing it.

**Comment by bkk87** [comment]
I did perform manual rescans as you suggested but it did not fix the problem for me.

**Comment by Cebrain** [comment]
> I did perform manual rescans as you suggested but it did not fix the problem for me.

As another workaround, you could test the memories app in the nextcloud store.
Its an alternative photo app for nextcloud and looks great. 
https://github.com/pulsejet/memories

**Comment by doits** [comment]
another one from me, hangs and crashes frequently (media tab is basically unusable)

### Cause of error
```java
Exception in thread "main" java.lang.IllegalStateException: Couldn't read row 659, col 3 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
    at android.database.CursorWindow.nativeGetString(Native Method)
    at android.database.CursorWindow.getString(CursorWindow.java:465)
    at android.database.AbstractWindowedCursor.getString(AbstractWindowedCursor.java:54)
    at android.database.CursorWrapper.getString(CursorWrapper.java:141)
    at com.owncloud.android.datamodel.FileDataStorageManager.createFileInstance(FileDataStorageManager.java:968)
    at com.owncloud.android.datamodel.FileDataStorageManager.getGalleryItems(FileDataStorageManager.java:2241)
    at com.owncloud.android.datamodel.FileDataStorageManager.getAllGalleryItems(FileDataStorageManager.java:2186)
    at com.owncloud.android.ui.adapter.GalleryAdapter.showAllGalleryItems(GalleryAdapter.kt:168)
    at com.owncloud.android.ui.fragment.GalleryFragment.showAllGalleryItems(GalleryFragment.java:380)
    at com.owncloud.android.ui.asynctasks.GallerySearchTask.onPostExecute(GallerySearchTask.java:126)
    at com.owncloud.android.ui.asynctasks.GallerySearchTask.onPostExecute(GallerySearchTask.java:48)
    at android.os.AsyncTask.finish(AsyncTask.java:771)
    at android.os.AsyncTask.-$$Nest$mfinish(Unknown Source:0)
    at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:788)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:201)
    at android.os.Looper.loop(Looper.java:288)
    at android.app.ActivityThread.main(ActivityThread.java:7872)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30230090`
* Build flavor: `gplay`

### Device information
* Brand: `google`
* Device: `cheetah`
* Model: `Pixel 7 Pro`
* Id: `TQ1A.221205.011`
* Product: `cheetah`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `9244662`

**Comment by AlvaroBrey** [comment]
This _may_ be fixed by #11098

**Comment by doits** [comment]
> This may be fixed by https://github.com/nextcloud/android/pull/11098

YES, I can confirm it! I downloaded the Test-Build from there and it is snappy as hell, no crashes! Thank you!

**Comment by AlvaroBrey** [comment]
Fixed by #11098
